src/documents/firstDiscoveryServer.md

```diff
@@ -0,0 +1,383 @@
+                Used to securely identify the client with the OAuth2.
+                Should be kept confidentially.
+                <p>
+                    Example:
+                    "client_secret_first_discovery"
+                </p>
+                <p>
+                    Option Path:
+                    <code>"preferencesConfig.authentication.client_secret"</code>
+                </p>
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### oauth2.json ###
+
+Builds on [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json), setting concrete values to configuration of the OAuth2 server that are known to exists for the GPII security layer.
+
+(see: [oauth2.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/oauth2.json))
+
+<table>
+    <thead>
+        <tr>
+            <th>Option Path</th>
+            <th>Description</th>
+            <th>Value</th>
+        <tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>
+                <code>"preferencesConfig.securityServer.paths.token"</code>
+            </td>
+            <td>
+                The path to the resource for requesting an access token
+            </td>
+            <td>
+                "/access_token"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.securityServer.paths.preferences"</code>
+            </td>
+            <td>
+                The path to the resource for creating a preference set.
+                A query parameter can be added to provide the view/ontology that the
+                preferences are to be stored in.
+            </td>
+            <td>
+                "/add-preferences?view=%view"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.authentication.grant_type"</code>
+            </td>
+            <td>
+                The grant type supported by the OAuth2 server.
+            </td>
+            <td>
+                "client_credentials"
+            </td>
+        </tr>
+        <tr>
+            <td>
+                <code>"preferencesConfig.authentication.scope"</code>
+            </td>
+            <td>
+                The level of permissions that are being requested.
+            </td>
+            <td>
+                "add_preferences"
+            </td>
+        </tr>
+    </tbody>
+</table>
+
+### vagrant.json ###
+
+Builds on [environment.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/environment.json) and [oauth2.json](https://github.com/GPII/first-discovery-server/tree/master/src/config/oauth2.json), setting concrete values to configuration of the OAuth2 server that are expected when being run in a  [Vagrant VM](https://www.vagrantup.com). The configuration assumes that an instance of the OAuth2 server is available through the host machine at port 8081.
```

Comment:
Similar problems with this sentence. "setting concrete values to configuration of the server that are expected".